plugins/module_utils/kibana.py

```diff
@@ -72,6 +73,76 @@ def get_alert_connector_type_by_name(self, connector_type_name):
     connector_types = self.send_api_request(endpoint, 'GET')
     return next(filter(lambda x: x['name'] == connector_type_name, connector_types), None)
   
+  def ensure_alert(self, method, alert_id=None):
+    endpoint = 'alerting/rule'
+    if method.upper() == 'PUT':
+      endpoint += f"/{alert_id}"
+    criteria = self.format_alert_conditions()
```

Comment:
Is it best to take an option here for the method, or we just assume that if an `alert-id` is getting passed then you need to use the alternate endpoint and method?

Reply:
Good point. I'll update the method behavior so that it only worries about the existence of `alert-id`